### ISSUE TITLE
EEXIST: file already exists - Workspace an obstacle?

### ISSUE THREAD

**Issue by NKarolak** [jimmymcp/al-test-runner#4]
Hi James,
I am trying to use your extension for the very first time, in my local repository running against a local Docker container.
Assuming my repo is located under "PATH\MyRepo", I am working with a workspace including subfolders "PATH\MyRepo\app" and "PATH\MyRepo\test". Needless to say; my tests are located within the test subfolder.

When I try to run my first test using "AL Test Runner - Run Current Test", I receive an error message:
> EEXIST: file already exists, mkdir 'PATH\MyRepo\app\\.altestrunner'

The .altestrunnner folder has been created in the app folder, not in the test folder.
If I open the test folder in VS Code, there is still no .altestrunner folder.

How to make it work?

**Comment by jimmymcp** [comment]
Hi,

If you are using a workspace with multiple roots then yes, I expect that that is the problem. The extension needs more work to allow you to specify which folder contains the extension with your tests.

I haven't been able to replicate your issue exactly, but I have got some inconsistent results with workspaces. Try dragging the folder that contains your test extension to be first in the workspace, if it isn't already and reloading VS Code.

**Comment by NKarolak** [comment]
Switching the folders within the workspance indeed puts the .altestrunner folder into the correct place, however it will still throw an error:

> EEXIST: file already exists, mkdir 'c:\Development\Repositories\DevOps\Product-Data-Management\test\\.altestrunner'

**Comment by NKarolak** [comment]
BTW I've tested to open the test folder seperately in VS Code. But the error persists there, too:

> EEXIST: file already exists, mkdir 'PATH\MyRepo\test\\.altestrunner'

**Comment by jimmymcp** [comment]
Hi Natalie,

I'm not sure what's happening here. I've built a new version of the app to output some values give us a clue. I'll send you an email.

**Comment by NKarolak** [comment]
The EEXIST error message was caused by comments in the `launch.json` (I had disabled the StartupObject configurations). The issue was not related to workspaces.

Next obstacles solved by:
- If you are running the bash terminal, "Select Default Shell" and choose PowerShell; then "Kill Terminal".
- Open PowerShell as administrator:
`install-module navcontainerhelper`  or `update-module navcontainerhelper ` 
`Check-NavContainerHelperPermissions -fix`
- in the `launch.json`, the servername must match exactly the docker container name (no trailing backslash, **case sensitive**)
- Retry, retry ... until a login window comes, then retry again.

Thanks for your help, James :)
